contrib/epee/src/net_ssl.cpp

```diff
@@ -130,9 +130,9 @@ bool create_ssl_certificate(std::string &pkey_buffer, std::string &cert_buffer)
   return success;
 }
 
-ssl_context_t create_ssl_context(const std::string &private_key_path, const std::string &certificate_path, std::list<std::string> allowed_certificates)
+ssl_context_t create_ssl_context(const std::string &private_key_path, const std::string &certificate_path, std::list<std::string> allowed_certificates, std::vector<std::vector<uint8_t>> allowed_fingerprints)
 {
-  ssl_context_t ssl_context({boost::asio::ssl::context(boost::asio::ssl::context::sslv23), std::move(allowed_certificates)});
+  ssl_context_t ssl_context{boost::asio::ssl::context(boost::asio::ssl::context::sslv23), std::move(allowed_certificates), std::move(allowed_fingerprints)};
```

Comment:
Can you expand on the distinction between those calls (ie, foo{} vs foo({})) ?

Reply:
Well, the original line basically invoked the constructor for ssl_context_t using the regular braces. This constructor only gets a single argument - because everything is joined using the curly braces. Looking at the available options, the only valid constructor here is one that takes an ssl_context_t, and since it is an unnamed temporary, it is rvalue qualified. So basically it creates the ssl_context_t and then moves it into another ssl_context_t. The first variant could this be written like this:

`ssl_context_t ssl_context(ssl_context_t{boost::asio::ssl::context(boost::asio::ssl::context::sslv23), std::move(allowed_certificates)});`

Now, the compiler is likely to eliminate this unnecessary extra object, so in practice this isn't a really big issue, but it doesn't add anything useful. Hence the second version, which just constructs the ssl_context_t.